src/ui/src/js/controllers/job_view.js

```diff
@@ -90,5 +94,81 @@ export default function jobViewController(
     loadJob();
   });
 
+  /*
+   * Execute the job service's "run ad hoc job" functionality.
+   */
+  function runAdHoc(jobId) {
+    const resettingInterval = $scope.resetTheInterval;
+
+    JobService.runAdHocJob(jobId, $scope.resetTheInterval).then(
```

Comment:
It probably doesn't matter, but you create the `resttingInteverval` constant above and then don't actually use it here.

Reply:
The function below closes over the environment established here.

Reply:
I agree it seems weird that you would set the constant and not pass it to the function. I wonder if that pattern makes the const act like a variable.